### PR TITLE
update check itemName

### PR DESCRIPTION
**Pull request by bourgeoa**

### CODE REVIEW

File: lib/handlers/put.js [hunk 1]
```diff
@@ -9,8 +9,15 @@ const { stringToStream } = require('../utils')
 async function handler (req, res, next) {
   debug(req.originalUrl)
   res.header('MS-Author-Via', 'SPARQL')
-
   const contentType = req.get('content-type')
+
+  // check if a folder or resource with same name exists
```
Comment (TallTed) [inline lib/handlers/put.js:14]:
```suggestion
  // check whether a folder or resource with same name exists
```

File: lib/ldp.js [hunk 2]
```diff
@@ -157,37 +157,13 @@ class LDP {
...
-      testName = container ? fs.lstatSync(testPath).isFile() : fs.lstatSync(testPath).isDirectory()
-    } catch (err) { testName = false }
-
-    if (testName) {
-      throw error(404, 'Container and resource cannot have the same name in URI')
-    }
 
-    // TODO: possibly package this in ldp.post
-    let resourceUrl = await ldp.getAvailableUrl(hostname, containerPath, { slug, extension })
+    // allways return a valid URL.
```
Comment (TallTed) [inline lib/ldp.js:161]:
```suggestion
    // always return a valid URL.
```

File: lib/ldp.js [hunk 3]
```diff
@@ -355,8 +327,9 @@ class LDP {
     } catch (err) { }
   }
 
+  // check if a document (or container) has the same name than a document (or container)
```
Comment (TallTed) [inline lib/ldp.js:330]:
```suggestion
  // check whether a document (or container) has the same name as another document (or container)
```

File: lib/ldp.js [hunk 4]
```diff
@@ -606,17 +579,26 @@ class LDP {
...
-    const { path: containerFilePath } = await this.resourceMapper.mapUrlToFile({ url: requestUrl })
-    let fileName = slug.endsWith(extension) || slug.endsWith(this.suffixAcl) || slug.endsWith(this.suffixMeta) ? slug : slug + extension
-    if (await promisify(fs.exists)(utilPath.join(containerFilePath, fileName))) {
-      fileName = `${uuid.v1()}-${fileName}`
-    }
-
-    return requestUrl + fileName
+    let itemName = slug.endsWith(extension) || slug.endsWith(this.suffixAcl) || slug.endsWith(this.suffixMeta) ? slug : slug + extension
+    try {
+      // check resource exists
```
Comment (TallTed) [inline lib/ldp.js:588]:
```suggestion
      // check whether resource exists
```

File: lib/ldp.js [hunk 5]
```diff
@@ -606,17 +579,26 @@ class LDP {
...
+      const context = container ? '/' : ''
+      await this.resourceMapper.mapUrlToFile({ url: (requestUrl + itemName + context) })
+      itemName = `${uuid.v1()}-${itemName}`
+    } catch (e) {
+      try {
+        // check resource with same name exists
```
Comment (TallTed) [inline lib/ldp.js:594]:
```suggestion
        // check whether resource with same name exists
```

**Review by TallTed — CHANGES_REQUESTED** [review]
small stuff, for humans

**Comment by bourgeoa** [comment]
> small stuff, for humans

Thanks. You are indeed very helpful for humans, that we still are ...

**Comment by bourgeoa** [comment]
merged with https://github.com/nodeSolidServer/node-solid-server/pull/1750

**Pull request closed without merging**